src/hooks/useFetch.js

```diff
@@ -0,0 +1,35 @@
+import { useState, useEffect } from "react";
+
+const useFetch = (url) => {
+  const [data, setData] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  const getFetchData = async (url) => {
+    if (isLoading === false || error !== null) {
+      setIsLoading(true);
+      setError(null);
+    }
+    try {
+      const response = await fetch(url);
+      if (!response.ok) {
+        throw new Error("HTTP Error");
+      }
+      const data = await response.json();
+      setData(data);
+      setIsLoading(false);
+    } catch (error) {
+      setData(null);
+      setError(error.message);
+      setIsLoading(false);
```

Comment:
you can also do the `setIsLoading` in a `finally` block https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/try...catch